### PR TITLE
[Improvement] Bump Flink CDC MySQL Connector version to 2.4.0 (#154)

### LINKED ISSUE #154: [Improvement] Bump Flink CDC MySQL Connector version to 2.4.0
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Flink CDC 2.4 has been released at Jul 25, which provides many new features, for example, automatically close idle readers at the end of the snapshot phase. I have already upgraded part of our online streaming ETL jobs based on Flink CDC 2.4 and Flink Doris Connector 1.4 framework, which look good so far. 
In this process, I solved some problems, and want to contribute my code to community.

### Solution

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by cygnusdark**
# Proposed changes

Issue Number: close #154 

## Problem Summary:

Bump Flink CDC MySQL Connector version to 2.4.0

## Checklist(Required)

1. Does it affect the original behavior: No
2. Has unit tests been added: No
3. Has document been added or modified: No
4. Does it need to update dependencies: Yes
5. Are there any changes that cannot be rolled back: No

## Further comments

I carefully tested and have re-deployed half of our online streaming ETL jobs using the newest Flink CDC 2.4 framework, which look good so far, except this issue: https://github.com/ververica/flink-cdc-connectors/issues/2243
The new feature of mysql connector that allows our job to automatically close idle readers has some problems when a single job syncs multiple tables and later we add new tables in sync list.
However, this feature looks good when you just want to sync single table. It will automatically release idle resources after snapshot reading done.
@JNSimba @DongLiang-0 anyone can help review this PR?

### CODE REVIEW

**Comment by JNSimba** [comment]
Thank you for your contribution, it has been upgraded to 2.4.1, please resolve the conflict

**Comment by cygnusdark** [comment]
The conflict has been resolved. Please check again.

File: flink-doris-connector/src/main/java/org/apache/doris/flink/tools/cdc/mysql/MysqlDatabaseSync.java [hunk 1]
```diff
@@ -21,6 +21,7 @@
 import com.ververica.cdc.connectors.mysql.source.config.MySqlSourceOptions;
 import com.ververica.cdc.connectors.mysql.source.offset.BinlogOffset;
 import com.ververica.cdc.connectors.mysql.source.offset.BinlogOffsetBuilder;
+import com.ververica.cdc.debezium.utils.JdbcUrlUtils;
```
Comment (JNSimba) [inline flink-doris-connector/src/main/java/org/apache/doris/flink/tools/cdc/mysql/MysqlDatabaseSync.java:24]: please remove this import

**Review by JNSimba — APPROVED** [review]
LGTM

**Pull request merged**